### PR TITLE
adapt new im_shape im Mask R-CNN-FPN

### PR DESCRIPTION
**Pull request by yghstill**

### CODE REVIEW

File: ppdet/data/transform/operator.py [hunk 1]
```diff
@@ -678,14 +678,14 @@ def apply(self, sample, context=None):
 
         im = self.apply_image(sample['image'], [im_scale_x, im_scale_y])
         sample['image'] = im
-        sample['im_shape'] = np.array([resize_h, resize_w], dtype=np.int32)
```
Comment (wangxinxin08) [inline ppdet/data/transform/operator.py:681]: why change array to asarray?

File: ppdet/data/transform/operator.py [hunk 2]
```diff
@@ -678,14 +678,14 @@ def apply(self, sample, context=None):
...
+        sample['im_shape'] = np.asarray([resize_h, resize_w], dtype=np.float32)
         if 'scale_factor' in sample:
             scale_factor = sample['scale_factor']
-            sample['scale_factor'] = np.array(
+            sample['scale_factor'] = np.asarray(
```
Comment (wangxinxin08) [inline ppdet/data/transform/operator.py:684]: same as above

File: ppdet/data/transform/operator.py [hunk 3]
```diff
@@ -678,14 +678,14 @@ def apply(self, sample, context=None):
...
                 [scale_factor[0] * im_scale_y, scale_factor[1] * im_scale_x],
                 dtype=np.float32)
         else:
-            sample['scale_factor'] = np.array(
+            sample['scale_factor'] = np.asarray(
```
Comment (wangxinxin08) [inline ppdet/data/transform/operator.py:688]: same as above

File: ppdet/data/transform/operator.py [hunk 4]
```diff
@@ -1408,9 +1408,9 @@ def apply(self, sample, context=None):
         canvas[:min(dim, resize_h), :min(dim, resize_w), :] = img[
             offset_y:offset_y + dim, offset_x:offset_x + dim, :]
         sample['image'] = canvas
-        sample['im_shape'] = np.array([resize_h, resize_w], dtype=np.int32)
+        sample['im_shape'] = np.asarray([resize_h, resize_w], dtype=np.float32)
```
Comment (wangxinxin08) [inline ppdet/data/transform/operator.py:1411]: same as above

File: ppdet/data/transform/operator.py [hunk 5]
```diff
@@ -1408,9 +1408,9 @@ def apply(self, sample, context=None):
...
         scale_factor = sample['sacle_factor']
-        sample['scale_factor'] = np.array(
+        sample['scale_factor'] = np.asarray(
```
Comment (wangxinxin08) [inline ppdet/data/transform/operator.py:1413]: same as above

File: ppdet/modeling/bbox.py [hunk 6]
```diff
@@ -104,7 +104,7 @@ def generate_proposal(self, inputs, rpn_head_out, anchor_out):
                 bbox_deltas=rpn_delta,
                 anchors=anchor,
                 variances=var,
-                im_info=inputs['im_info'],
```
Comment (jerrywgz) [inline ppdet/modeling/bbox.py:107]: 这个改动影响训练了

Comment (yghstill) [reply ppdet/modeling/bbox.py:107]: Done,thanks

File: ppdet/data/transform/operator.py (hunk 1 above)
Comment (yghstill) [reply ppdet/data/transform/operator.py:681]: 两者没有区别，只在特定情况下减少copy次数

**Review by jerrywgz — APPROVED** [review]

**Pull request merged**